src/ssl.c

```diff
@@ -44016,6 +44248,32 @@ int wolfSSL_i2d_X509_NAME(WOLFSSL_X509_NAME* name, unsigned char** out)
         return (ret == 0) ? WOLFSSL_SUCCESS : WOLFSSL_FAILURE;
     }
 
+    /**
+     * wolfSSL_add1_chain_cert makes a copy of the cert so we free it
+     * on success
+     */
+    int wolfSSL_add0_chain_cert(WOLFSSL* ssl, WOLFSSL_X509* x509)
+    {
+        WOLFSSL_ENTER("wolfSSL_add0_chain_cert");
+        WOLFSSL_MSG("WARNING: This function modifies the certs used by all "
+                    "SSL objects created using the same CTX");
+        if (wolfSSL_add1_chain_cert(ssl, x509) != WOLFSSL_SUCCESS) {
+            return WOLFSSL_FAILURE;
+        }
+        wolfSSL_X509_free(x509);
+        return WOLFSSL_SUCCESS;
+    }
+
+    int wolfSSL_add1_chain_cert(WOLFSSL* ssl, WOLFSSL_X509* x509)
+    {
+        WOLFSSL_ENTER("wolfSSL_add1_chain_cert");
+        WOLFSSL_MSG("WARNING: This function modifies the certs used by all "
+                    "SSL objects created using the same CTX");
+        if (ssl == NULL || ssl->ctx == NULL)
+            return WOLFSSL_FAILURE;
+        return wolfSSL_CTX_add1_chain_cert(ssl->ctx, x509);
```

Comment:
Do we actually want to change the context?

Reply:
We don't currently support per SSL chains. I can look into how difficult this would be. Alternatively, this could be a stub that doesn't do anything/errors out.

Reply:
Please make a change here or give an explanation as to why not.

Reply:
Changed in https://github.com/wolfSSL/wolfssl/pull/3726/commits/f440b406c147772e9d7b17e5ebad987b67bd0e84 to not affect the context.